engine/pom.xml

```diff
@@ -218,8 +218,10 @@
                             org.hibernate.search.query;version="${project.version}",
                             org.hibernate.search.query.collector;version="${project.version}",
                             org.hibernate.search.query.dsl;version="${project.version}",
+                            org.hibernate.search.query.dsl.sort;version="${project.version}",
                             org.hibernate.search.query.engine;version="${project.version}",
                             org.hibernate.search.query.engine.spi;version="${project.version}",
+                            org.hibernate.search.query.engine.spi.impl;version="${project.version}",
```

Comment:
It's weird indeed. In this case, it comes from the fact that the Lucene backend implementation is bundled with the Hibernate Search engine, so I can't do much to prevent this, apart from renaming the package (which would just hide the weirdness in my opinion, not solve the actual problem).
On 6.0, if we actually have an API free of Lucene dependencies, we may be able to split things so that the Lucene backend has its own separate module, and this weirdness would disappear. But that's way out of the scope of this PR, I'm afraid.

On a side note, there already is other packages with SPI implementations provided by the "engine" module. See for instance `org.hibernate.search.cfg.spi.impl` or `org.hibernate.search.spi.impl`.